### PR TITLE
fix redis worker hang when redis connection is lost or redis server is restarted

### PR DESCRIPTION
**Pull request by starplanet**
currently if I restart redis, redis worker will report connection error and then run_forever loop will stop.

### CODE REVIEW

File: service_streamer/service_streamer.py [hunk 1]
```diff
@@ -478,9 +489,13 @@ def send_request(self, task_id, request_id, model_input):
         self._redis.lpush(self._redis_request_queue_name, pickle.dumps(message))
 
     def recv_response(self, timeout):
-        message = self._response_pb.get_message(timeout=timeout)
-        if message:
-            return pickle.loads(message["data"])
+        try:
+            message = self._response_pb.get_message(timeout=timeout)
+            if message:
```
Comment (Meteorix) [inline service_streamer/service_streamer.py:494]: 建议这个return放到try else里面

File: service_streamer/service_streamer.py [hunk 2]
```diff
@@ -478,9 +489,13 @@ def send_request(self, task_id, request_id, model_input):
...
+                return pickle.loads(message["data"])
+        except RedisError as e:
+            logger.warning('recv_response: redis error occurs in get_message, %s', str(e))
+            self._setup()  # 重新订阅
```
Comment (Meteorix) [inline service_streamer/service_streamer.py:498]: 重新订阅之后，还能继续work吗？

**Comment by Meteorix** [comment]
@starplanet thanks for your pr. I have 2 comments, please take a look.

File: service_streamer/service_streamer.py (hunk 2 above)
Comment (starplanet) [reply service_streamer/service_streamer.py:498]: 可以验证过了

File: service_streamer/service_streamer.py (hunk 1 above)
Comment (starplanet) [reply service_streamer/service_streamer.py:494]: 放到else里面，有什么好处吗？

Comment (Meteorix) [reply service_streamer/service_streamer.py:494]: 代码逻辑更清晰，return这一句本身是不需要try/catch的

**Pull request closed without merging**